app/src/main/java/org/mozilla/fenix/onboarding/HomeOnboardingDialogFragment.kt

```diff
@@ -4,48 +4,72 @@
 
 package org.mozilla.fenix.onboarding
 
+import android.annotation.SuppressLint
+import android.content.pm.ActivityInfo
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
 import androidx.fragment.app.DialogFragment
+import androidx.navigation.fragment.findNavController
+import com.google.accompanist.insets.ProvideWindowInsets
+import mozilla.components.lib.state.ext.observeAsComposableState
 import org.mozilla.fenix.R
-import org.mozilla.fenix.databinding.FragmentOnboardingHomeDialogBinding
+import org.mozilla.fenix.components.components
+import org.mozilla.fenix.ext.nav
 import org.mozilla.fenix.ext.settings
+import org.mozilla.fenix.onboarding.view.Onboarding
+import org.mozilla.fenix.theme.FirefoxTheme
 
 /**
- * Dialog displayed once when one or multiples of these sections are shown in the home screen
- * recentTabs,recentBookmarks,historyMetadata or pocketArticles.
+ * Dialog displaying a welcome and sync sign in onboarding.
  */
 class HomeOnboardingDialogFragment : DialogFragment() {
+    @SuppressLint("SourceLockedOrientationActivity")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setStyle(STYLE_NO_TITLE, R.style.HomeOnboardingDialogStyle)
+        activity?.requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
```

Comment:
My conversation with Kate was that we didn't want the dialog to have a landscape mode. So, this is to just lock it to portrait mode.